scripts/changelog.ts

```diff
@@ -0,0 +1,380 @@
+};
+
+/**
+ * Creates a new git tag based on a version string
+ *
+ * @param version - Version to name tag
+ */
+function createTag(version: string): void {
+	child.spawnSync("git", ["commit", "-am", `Release ${version}`]);
+	child.spawnSync("git", ["tag", version]);
+	child.spawnSync("git", ["push"]);
+	child.spawnSync("git", ["push", "origin", `v${version}`]);
+}
+
+/**
+ * Transform a list of commit objects into a map of version names to commits
+ *
+ * @param commits - Commit list to transform
+ * @param currentVersion - Version used to categorize commits since the last tag
+ * @returns - Map of version names to commits within a version
+ */
+function createTagMap(
+	commits: Array<Commit>,
+	currentVersion: string,
+): Record<string, Array<Commit>> {
+	const versionMap: Record<string, Array<Commit>> = {
+		[currentVersion]: [],
+	};
+	let currentTag = currentVersion;
+
+	commits.forEach((commit) => {
+		const tagMatch =
+			commit.refNames && commit.refNames.match(/tag: v\d+\.\d+\.\d+/);
+		const tagName = tagMatch && tagMatch[0].replace(/^tag: /, "");
+		currentTag = tagName || currentTag;
+		if (!currentTag) {
+			return;
+		}
+		versionMap[currentTag] = [...(versionMap[currentTag] || []), commit];
+	});
+
+	return versionMap;
+}
+
+/**
+ * Generates changelog markdown from a map of tags
+ * @param tagMap - Map of version names to commits within them
+ * @returns - Markdown string
+ */
+function generateMarkdown(tagMap: Record<string, Array<Commit>>): string {
+	function renderItems(items: Array<Commit>, title: string) {
+		return items.length > 0
+			? `## ${title}\n\n${items.map((commit) =>
+					`- ${commit.subject ? commit.subject.trim() : ""}`
+				).join("\n")}\n`
+			: "";
+	}
+	return `
+# Changelog
+
+All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.
+${Object.keys(tagMap).map((tag) => {
+		const commits = tagMap[tag];
+		const features = commits.filter(({meta}) => meta?.commitType === "feat");
+		const fixes = commits.filter(({meta}) => meta?.commitType === "fix");
+		const breaking = commits.filter(({meta}) => meta?.breaking);
+		const misc = commits.filter(({meta}) => !meta?.breaking && meta?.custom);
+		return `
+## [${tag}](https://github.com/romefrontend/rome/releases/tag/${tag})
+
+${renderItems(features, "Features")}
+${renderItems(fixes, "Bug fixes")}
+${renderItems(breaking, "BREAKING CHANGES")}
+${renderItems(misc, "Miscellaneous")}`;
+	}).join("\n")}
+\n\n\n\n\n
+<img alt="Rome, logo of an ancient Greek spartan helmet" src="https://github.com/romefrontend/rome/raw/main/assets/PNG/logomark_transparent.png" width="128px">
+`;
+}
+
+/**
+ * Get the current package version
+ *
+ * @returns - Promise resolving to the current version
+ */
+async function getCurrentVersion(): Promise<string> {
+	return String(
+		JSON.parse(await readFileText(ROOT.append("package.json"))).version,
+	);
```

Comment:
I don't think this is necessarily more readable but it would produce some nicer errors:

```suggestion
	const path = ROOT.append("package.json");
	return consumeJSON({input: await readFileText(path), path}).get("version").asString();
```